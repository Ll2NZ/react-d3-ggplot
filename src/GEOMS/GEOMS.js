import React            from "react";
import PropTypes        from "prop-types";
import { ChartContext } from "../_context/";

function GEOMS(props) {
        return(
            <ChartContext.Provider value={ props }>
                <svg width={ props.dimensions.width } height={ props.dimensions.height }>
                    { props.children }
                </svg>
            </ChartContext.Provider>
        );
}

GEOMS.propTypes = {
    dimensions: PropTypes.object,
    children: PropTypes.any
};

export default GEOMS;
