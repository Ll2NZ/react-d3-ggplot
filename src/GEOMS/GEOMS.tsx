import React, { ReactChildren, ReactComponentElement, ReactChild } from "react";
import { ChartContext, IContext } from "../_context/";

interface IProps extends IContext {
  children?: ReactChild[] | ReactChild;
}

function GEOMS(props: IProps) {
  if (!props.data || !props.aes || !props.dimensions) {
    throw new Error(`GEOMS must have the following properties: data, aes, and dimension`);
  }

  if (!props.children) {
    throw new Error("GEOMS expects at least one child components such as: <Rects/> or <Points/>");
  }

  return (
    <ChartContext.Provider value={props}>
      <svg width={props.dimensions.width} height={props.dimensions.height}>
        {props.children}
      </svg>
    </ChartContext.Provider>
  );
}

export default GEOMS;
