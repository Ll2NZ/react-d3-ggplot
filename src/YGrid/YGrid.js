import React              from "react";
import { Component }      from "react";
import PropTypes          from "prop-types";
import { ScalesConsumer } from "../Context/";
import { select }         from "d3-selection";
import { axisLeft }     from "d3-axis";

class YGrid extends Component {
    static contextType = ScalesConsumer;

    static propTypes = {
        createScaleType: PropTypes.func
    }

    createYGridLines(){
        // create new yScale
        const { aes } = this.context;
        const yScale  = this.props.createScaleType(aes[1], this.constructor.name);

        // find location of grid lines
        const { dimensions } = this.context;
        const axisPosition   = `translate(${dimensions.padding}, 0)`;

        // append YGrid lines
        select(this.node)
            .attr("class", "y-grid")
            .attr("transform", axisPosition)
            .call(
                axisLeft(yScale)
                .ticks()
                .tickSize(-(dimensions.width - dimensions.padding*2))
                .tickFormat("")
            );

        // color lines
        select(".y-grid")
            .selectAll("line")
            .attr("stroke", "rgb(255, 255, 255)");

        // get rid of black color overlaying y-axis
        select(".y-grid")
            .select("path")
            .attr("stroke", "rgb(255, 255, 255)");
    }

    render(){
        return(
            <g
                ref={ node => this.node = node }
            >
            </g>
        );
    }

    componentDidMount(){
        this.createYGridLines();
    }

    componentDidUpdate(){
        this.createYGridLines();
    }
}

export default YGrid;
