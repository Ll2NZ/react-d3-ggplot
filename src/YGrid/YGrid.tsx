import React from "react";
import { useContext } from "react";
import { useRef } from "react";
import { useEffect } from "react";
import { useScale } from "../_hooks/";
import { ChartContext } from "../_context/";
import { select } from "d3-selection";
import { axisLeft } from "d3-axis";

interface IProps {}

function YGrid(props: IProps) {
  const node: any = useRef(null);
  const context = useContext(ChartContext);
  const scale: any = useScale(context, YGrid.name);

  return <g ref={node} />;
}

export default YGrid;
