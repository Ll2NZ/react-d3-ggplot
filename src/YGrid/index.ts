export { default as YGrid } from "./YGrid";
