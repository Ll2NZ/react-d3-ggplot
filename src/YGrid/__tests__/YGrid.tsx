import React from "react";
import { render, cleanup } from "@testing-library/react";
import { GEOMS } from "../../GEOMS/";
import YGrid from "../YGrid";

afterEach(cleanup);

const validState = {
  data: [{ x: 0, y: 0 }, { x: 10, y: 10 }],
  aes: ["x", "y"],
  dimensions: { width: 500, height: 400, padding: 50 }
};

test("Should render multiple <line> elements", () => {
  const node = render(
    <GEOMS {...validState}>
      <YGrid />
    </GEOMS>
  );
});

test("Underlyng markup should not change as we change our logic in code base unless we chage the markup ourselves", () => {
  const { container } = render(
    <GEOMS {...validState}>
      <YGrid />
    </GEOMS>
  );

  expect(container.firstChild).toMatchSnapshot();
});
