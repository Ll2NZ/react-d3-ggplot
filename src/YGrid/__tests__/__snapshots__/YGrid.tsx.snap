// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Underlyng markup should not change as we change our logic in code base unless we chage the markup ourselves 1`] = `
<svg
  height="400"
  width="500"
>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="start"
    transform="translate(50, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,350.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,290.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,230.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,170.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,110.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,50.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="400"
      />
    </g>
  </g>
</svg>
`;
